app/controllers/api/v1/users/descriptions_controller.rb

```diff
@@ -0,0 +1,16 @@
+class Api::V1::Users::DescriptionsController < Api::V1::BaseController
+  
+  def create
+    describee = User.find(params[:id])
+    descriptions = JSON.parse(request.body.read)
+    describer    = User.find(descriptions['describer_id'])
+    describee.create_descriptions(descriptions['johari'], describer)
+    if describee.all_descriptions_completed?(describer.id, descriptions['johari'])
+      Assignment.complete(describee.id, describer.id)
```

Comment:
Yeah that shouldn't be too hard. just have to find the assignments where completed? is false